src/ranged.cpp

```diff
@@ -161,19 +161,58 @@ bool player::handle_gun_damage( item &it, int shots_fired )
 {
     if( !it.is_gun() ) {
         debugmsg( "Tried to handle_gun_damage of a non-gun %s", it.tname() );
+        shots_fired = shots_fired + 0; // left this here in case someone needs shots_fired in the future.
         return false;
     }
 
     const auto &curammo_effects = it.ammo_effects();
     const cata::optional<islot_gun> &firing = it.type->gun;
+    const std::string thistimefired = to_string_time_of_day(calendar::turn);
+    // Here we check for a chance for the weapon to suffer a misfire due to
+    // being dirty (blackpowder fouling effect only; typical dirt accumulation
+    // from smokeless powder is not currently modeled by this effect).
+    // Can damage gun, but not enough to push the weapon beyond 'shattered'.
+    // Check, but not accumulation, will happen even when firing normal rounds,
+    // so you can screw up your MP5 by firing blackpowder rounds and the
+    // effects will be in force even when firing retail ammo, though the gun
+    // won't get MORE foul.
+
+    // causes failure to cycle if weapon is fired too quickly:
+    if (it.last_fired == thistimefired && ( curammo_effects.count("MUZZLE_SMOKE") || curammo_effects.count("BLACKPOWDER") ) ) {
+        add_msg_player_or_npc(_("Your %s fails to cycle!"),
+            _("<npcname>'s %s fails to cycle!"),
+            it.tname());
+            // Don't increment until after the message
+        return false;
+    }
+    it.last_fired = thistimefired;
+    // chance to cause malfunction, which always happens when clogged:
+    if( ( it.dirt > 100 && one_in( ( ( 501 - it.dirt ) ) / 40 ) ) ||
+        ( it.has_fault( fault_gun_clogged ) ) ) {
+        if( !it.has_fault( fault_gun_clogged ) ) {
+            it.faults.insert( fault_gun_clogged );
+        }
+        if( it.has_fault( fault_gun_blackpowder ) ) {
+            it.faults.erase( fault_gun_blackpowder );
+        }
+        add_msg_player_or_npc( _( "Your foul %s misfires with a muffled click!" ),
+                               _( "<npcname>'s foul %s misfires with a muffled click!" ),
+                               it.tname() );
+        // chance to damage gun:
+        if( it.damage() < it.max_damage() && it.dirt > 350 && one_in( ( ( 505 - it.dirt ) ) / 30 ) ) {
```

Comment:
As I got told, blackpowder fouling doesn't actually damage the gun if you clean it quickly after firing. It may indeed affect accuracy and reliability, but if you want to simulate that, better just calculate penalties to those from current "dirt" value of the gun.

Reply:
The code I made doesn't cause damage by "fouling" alone, it causes damage sometimes when the gun is fired. This is due to parts being stressed by incorrect (unexpected) powder loads, incomplete cycling, etc...

Reply:
See issue #31553 ; over pressure ammunition is absolutely a concern, but the danger is less with blackpowder. Unless a round's pressure is so low that it squibs (fails to exit the barrel), the primary source of damage in blackpowder munitions will be rusting. Squibbing with Blackpowder will be hard to due- great volumes are needed to get near smokeless energies; therefore it's very hard to over/underload a case.

The residues are highly hygroscopic. Small parts like extractors, springs, perhaps firing pin tips, could, in theory, receive too much rusting over time, but it would take a long time for even this amount of damage to be seen due to accumulated water.

What's more likely is the firearm simply refusing to go into battery (Locked breech firearms whose locking surfaces are occluded due to fouling, or blowback firearms where the boltgap are made too great), and thus refusing to fire, or small diameter gas tubes/ ports being filled too much with fouling and failing to cycle the weapon. 

Damage due to blackpowder is most likely going to occur due to an out of battery detonation in firearms (read homemade, crappy, some blowback firearms) that lack out of battery safeties. Most firearms have out of battery safeties, though blowback ones tend not to as the damage potential from an OOB detonation is less than that of a locked breach firearm.


TLDR: blackpowder fouling isn't likely to damage a gun unless it's homemade, crappy or some form of blowback, and simple blowback firearms will receive less damage than delayed/ locked breech firearms.